docs/operating/linux.md

```diff
@@ -135,4 +137,21 @@ If you wish to change those, you are expected to understand those implications a
 
 The service was created assuming it'll be used in a production scenario.
 
-In case you want to use this service for development as well, you may need to adjust the `ExecStart` line to include the `--development` flag if your development environment doesn't support Direct IO, or if you require smaller cache sizes and/or batch sizes due to memory constraints.
\ No newline at end of file
+In case you want to use this service for development as well, you may need to adjust the `ExecStart` line to include the `--development` flag if your development environment doesn't support Direct IO, or if you require smaller cache sizes and/or batch sizes due to memory constraints.
+
+### Memory Locking
+
+TigerBeetle requires `RLIMIT_MEMLOCK` to be set high enough to:
+
+1. initialize io_uring, which requires memory shared with the kernel to be locked, as well as
+2. lock all allocated memory, and so prevent the kernel from swapping any pages to disk, which would not only affect performance but also bypass TigerBeetle's storage fault-tolerance.
+
+If the required memory cannot be locked, then the environment should be modified either by (in order of preference):
+
+1. giving the local `tigerbeetle` binary the `CAP_IPC_LOCK` capability, or
```

Comment:
We could suggest the exact command: `sudo setcap "cap_ipc_lock=+ep" tigerbeetle`

Reply:
(additionally, we could suggest it in the log warning message directly too.)